src/cephadm/cephadmlib/templating.py

```diff
@@ -29,6 +29,34 @@ def __repr__(self) -> str:
         return repr(self.value)
 
 
+class TemplateNotFoundInZipApp(jinja2.TemplateNotFound):
+    def __init__(
+        self,
+        template: str,
+        *,
+        path: str = '',
+        relative_path: str = '',
+        archive_norm_path: str = '',
+        archive_path: str = ''
+    ) -> None:
+        super().__init__(template)
+        self.path = path
+        self.relative_path = relative_path
+        self.archive_norm_path = archive_norm_path
+        self.archive_path = archive_path
+
+    def __str__(self) -> str:
+        msg = self.message
+        msg2 = ''
+        if self.path or self.relative_path:
+            msg2 += f' path [{self.path!r}, rel={self.relative_path!r}] not found'
+        if self.archive_norm_path or self.archive_path:
+            msg2 += f' in [{self.archive_norm_path!r}, orig={self.archive_path!r}]'
+        if msg2:
+            msg2 = ':' + msg2
+        return f'{msg}{msg2}'
```

Comment:
(nit) this might be easier to read if we remove the various if clauses and simply return an f-string